settings.go

```diff
@@ -1,5 +1,101 @@
 package main
 
+import (
+	"github.com/deckarep/golang-set"
+	"github.com/kubewarden/gjson"
+
+	"fmt"
+)
+
+type Settings struct {
+	RequireTls bool       `json:"requireTLS"`
+	AllowPorts mapset.Set `json:"allowPorts"`
+	DenyPorts  mapset.Set `json:"denyPorts"`
+}
+
+// Builds a new Settings instance starting from a validation
+// request payload:
+// {
+//    "request": ...,
+//    "settings": {
+//       requireTLS: ...
+//    }
+// }
+func NewSettingsFromValidationReq(payload []byte) (Settings, error) {
+	// Note well: we don't validate the input JSON now, this has
+	// already done inside of the `validate` function
+
+	return newSettings(
+		payload,
+		"settings.requireTLS",
+		"settings.allowPorts",
+		"settings.denyPorts")
+}
+
+// Builds a new Settings instance starting from a Settings
+// payload:
+// {
+//    "requireTLS": ...
+// }
+func NewSettingsFromValidateSettingsPayload(payload []byte) (Settings, error) {
+	if !gjson.ValidBytes(payload) {
+		return Settings{}, fmt.Errorf("invalid JSON payload")
+	}
+
+	return newSettings(
+		payload,
+		"requireTLS",
+		"allowPorts",
+		"denyPorts")
```

Comment:
Nitpick: this duplication here and on `NewSettingsFromValidationReq` seems a bit weird now that we are making the example fancy, I think we could probably abstract this list somewhere else, with a prefix argument (either none or `settings.`).

Reply:
I had to resort to this duplication because:
  * The gjson queries change, depending on whether the payload is coming from the `validate` or the `validate_settings` function
  * I wanted to ensure the JSON validation is done only once: building a `Settings` instance from the `validate` payload doesn't require a validation check, because this is done inside of the `validation` function.

Reply:
The suggestion was just to define the list once. The caller then can pass an argument to the list retrieval function that prefixes every element. This way the list itself is only written once, but the caller decides what prefix (if any) to use.

If a new setting field is added, only one place has to be modified.